### PR TITLE
Add rule enforcing that certain package.json fields must have a provided rationale

### PR DESCRIPTION
**Pull request by jofrey-a-msft-production**

### CODE REVIEW

File: packages/rules-core/src/enforce-rationale.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          asBollLineNumber(0),
+          `The "fields" option for rule ${ruleName} should be an array`
+        )
+      ];
+    }
+
+    const fields = this.options.fields as string[];
+    const contents = JSON.parse(file.content);
+
+    if (fields.length && !contents.rationale) {
```
Comment (jdhuntington) [inline packages/rules-core/src/enforce-rationale.ts:39]: I think this should only happen if `contents` actually declares the field(s) that need rationale but doesn't include the rationale. 

ie, this should be ok:

```json
{
  "name": "foo"
}
```

config:
```js
const options = {
  fields: ['someBadKeyUsage']
```

File: packages/recommended/src/index.ts [hunk 2]
```diff
@@ -26,6 +26,10 @@ export const bootstrapRecommendedConfiguration = () => {
   RuleRegistryInstance.register("RedundantImportsDetector", () => new RedundantImportsDetector());
   RuleRegistryInstance.register("ESLintPreferConstRule", (l: Logger) => new ESLintPreferConstRule(l));
   RuleRegistryInstance.register("NoRedundantDepsRule", (l: Logger) => new NoRedundantDepsRule(l));
+  RuleRegistryInstance.register(
+    "EnforceRationale",
+    (l: Logger, options?: EnforceRationaleOptions) => new EnforceRationale(l, options)
```
Comment (jdhuntington) [inline packages/recommended/src/index.ts:31]: I don't think `options` will ever be populated. Probably shouldn't be in this callback signature.

File: packages/rules-core/src/enforce-rationale.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    if (extname(file.filename) !== ".json") {
+      return [new Success(ruleName)];
+    }
+
+    if (!this.options) {
+      return [new Success(ruleName)];
+    }
+
+    const failures: Result[] = [];
+    const fileKey = file.filename.slice(process.cwd().length + 1);
```
Comment (jdhuntington) [inline packages/rules-core/src/enforce-rationale.ts:31]: I feel like `filename.slice(process.cwd().length + 1)` should be a method on `file` called 'relativeFilename` or something similar.

**Review by jdhuntington — APPROVED** [review]

File: packages/recommended/src/index.ts (hunk 2 above)
Comment (jofrey-a-msft-production) [reply packages/recommended/src/index.ts:31]: Is there currently a way to plumb through options specified in the boll config for a particular rule to the instantiation of the rule itself? That's what I'd like to be doing here but I wasn't certain if there was a method for doing that just yet.

**Pull request merged**